lightningd/json.c

```diff
@@ -129,6 +130,16 @@ void json_add_address(struct json_result *response, const char *fieldname,
 		json_add_string(response, "address", addrstr);
 		json_add_num(response, "port", addr->port);
 	}
+	else if (addr->type == ADDR_TYPE_TOR_V2) {
+		json_add_string(response, "type", "torv2");
+		json_add_string(response, "address", tal_fmt(NULL, "%.22s", fmt_wireaddr_without_port(NULL, addr)));
```

Comment:
Use `tal_fmt(tmpctx, ...)` instead of `tal_fmt(NULL`, otherwise this will be a leak.  Similarly `fmt_wireadddr_without_port`, use `tmpctx` as context instead.

Also, is not `fmt_wireaddr_without_port` enough?  Why allocate twice as many strings?

Reply:
@ZmnSCPxj  will try this on next commit, had some issues with ctx but now we have g tmpctx so this should work then with a single call ;-)

Reply:
done ;-)